packages/server/src/db/invite-codes.ts

```diff
@@ -0,0 +1,31 @@
+import { Entity, Column, PrimaryColumn } from 'typeorm'
+
+@Entity({ name: 'invite_codes' })
+export class InviteCode {
+  @PrimaryColumn('varchar')
+  code: string
+
+  @Column('integer')
+  availableUses: number
+
+  @Column({ type: 'boolean', default: false })
+  disabled: boolean
+
+  @Column('varchar')
+  forUser: string
+
+  @Column('varchar')
+  createdBy: string
+}
+
+@Entity({ name: 'invite_codes_uses' })
+export class InviteCodeUses {
+  @PrimaryColumn('varchar')
+  code: string
+
+  @PrimaryColumn('varchar')
```

Comment:
Hey was just doing some early review and noticed you've got multiple PrimaryColumns here, is that correct?

Reply:
here it is yup.
We have the `InviteCode` table where the code is unique. And then the `InviteCodeUse` table where each code can have more than one use, but only one for each user. So the primary key is code + user_that_used_that_code

Reply:
Ah I was unclear on whether you _could_ have multiple primary columns on a table. It looks like TypeORM turns that into a composite primary key.

Reply:
ah yupp it does^^